content/2015.11.shweta-oak.rst

```diff
@@ -0,0 +1,108 @@
+shared with another user in order to collaborate and share data between multiple
+users on the same Kinto server. 
+
+To solve this problem, the following approach can be taken:
+
+Let the user enter the location of its Kinto instance, the first time.
+======================================================================
+
+Add a way, on client side, to specify where the server is. For instance, this should
+be done in JavaScript and HTML on  top of the Kinto.js client.
+It is basically a user interface for how to specify the "remote server" to use. The
+user has to enter the location of the server only once, the first time he uses a
+Kinto instance, later on, this location will be discovered.
+
+The best way to start doing this is to do it in a webpage that uses Kinto.js and let
+the user specify the server location. Once we have this, we will need to make this
+project re-usable by others.
+
+Store this location remotely
+============================
+
+Then, it would be good if we were able, for a given authenticated user, to automatically
+discover where is their Kinto instance is located. To do that we will need to find a solution
+to save this location somewhere when the user enters the location of the server for the first
+time and then, the information will be discovered.
+
+We also need to keep in mind that users may have multiple Kinto instances with regards to the
+application they will be using. The discovery of Kinto will be cross application too.
+
+We will now need to store this information remotely. This is where the ideas of a central
+repository and of a Distributed Hash Table (DHT) come into play. Since a DHT is by far more
+complex than a central repository, we will start with the use of a central server.
+
+
+(B1) Central Repository:
+
+A small Javascript helper to lookup the location of the server remotely once the user is connected. The flow would then look like this:
+
+        1. Connect to your account (Github, Twitter, FxA) using OAuth.
+        2. The client fetches the location of the Kinto server on a central Kinto collection using the user credentials.
+        3. The Kinto.js collection is configured to use this remote.
+
+pros of using a central repository:
+Its really easy to setup and implement. The application searches for the user’s server by looking up the central repository.
+
+cons of using a central repository:
+The user needs to trust a 3rd party as the links to the information of the user is being stored in one place.
+There is a possibility of leak of information that someone is connected to a Kinto instance, and where this instance is.  
+
+(B2) Distributed Hash Table (DHT)
+
+Once we have this mechanism in place, we can expand it in order to rely on a DHT rather than on a central repositories. DHT is a decentralized distributed system that provides a lookup service similar to a hash table. Different options can be used here (Matrix.org is one).
```

Comment:
Matrix.org is not a DHT so we should just remove the comment within parenthesis.